app/models/mixins/retirement_mixin.rb

```diff
@@ -231,15 +231,34 @@ def valid_zone?
     respond_to?(:my_zone) && my_zone.present?
   end
 
-  def setup_event_hash(requester)
-    event_hash = {:retirement_initiator => "system"}
-    event_hash[retirement_base_model_name.underscore.to_sym] = self
-    event_hash[:host] = host if self.respond_to?(:host)
+  def system_context_requester
+    if try(:evm_owner_id)
+      User.find(evm_owner_id)
+    else
+      $log.info("System context defaulting to admin user because owner of #{name} not set.")
+      User.in_my_region.find_by(:userid => 'admin')
```

Comment:
```suggestion
      User.super_admin
```

Please be aware when you are hardcoding an id like "admin"
Chances are we should not do that.